### PR TITLE
8253475: Javadoc clean up in HttpExchange and HttpServer

### PR DESCRIPTION
**Pull request by pconcannon**
Hi,

Could someone please review my doc-only fix for JDK-8253475 - 'Javadoc clean up in HttpExchange and HttpServer' ?

This fix is set of formatting changes intended to clean up the javadoc of the following classes :

`com.sun.net.httpserver.HttpExchange`
`com.sun.net.httpserver.HttpServer`

This issue is a sub-task of [JDK-8252822](https://bugs.openjdk.java.net/browse/JDK-8252822)

Kind regards,
Patrick

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- |
| Build | ✔️ (3/3 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) | ✔️ (9/9 passed) | ❌ (1/9 failed) | ✔️ (9/9 passed) |

**Failed test task**
- [Windows x64 (langtools/tier1)](https://github.com/pconcannon/jdk/runs/1216337098)

### Issue
 * [JDK-8253475](https://bugs.openjdk.java.net/browse/JDK-8253475): Javadoc clean up in HttpExchange and HttpServer

### Reviewers
 * [Daniel Fuchs](https://openjdk.java.net/census#dfuchs) (@dfuch - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/506/head:pull/506`
`$ git checkout pull/506`

### CODE REVIEW

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java [hunk 1]
```diff
@@ -36,30 +36,36 @@
...
- * <li>{@link #getResponseBody()} to get a {@link java.io.OutputStream} to send the response body.
- *      When the response body has been written, the stream must be closed to terminate the exchange.
+ *
+ * <p> The typical life-cycle of a {@code HttpExchange} is shown in the sequence
+ * below:
+ * <ol>
+ *     <li>{@link #getRequestMethod()} to determine the command.
+ *     <li>{@link #getRequestHeaders()} to examine the request headers (if
+ *     needed).
+ *     <li>{@link #getRequestBody()} returns a {@link java.io.InputStream} for
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java:46]: Hi Patrick, could you double check what the text of the link is in the generated javadoc here? Is it a `java.io.InputStream` or just a `InputStream` (in which case `a` should be `an`)

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java [hunk 2]
```diff
@@ -68,206 +74,227 @@
...
-     * {@link #getResponseBody()} will return the given
-     * {@link java.io.OutputStream}. The streams provided to this
-     * call must wrap the original streams, and may be (but are not
-     * required to be) sub-classes of {@link java.io.FilterInputStream}
-     * and {@link java.io.FilterOutputStream}.
-     * @param i the filtered input stream to set as this object's inputstream,
-     *          or {@code null} if no change.
-     * @param o the filtered output stream to set as this object's outputstream,
-     *          or {@code null} if no change.
+     * Used by {@linkplain #Filter Filters} to wrap either (or both) of this exchange's InputStream
```
Comment (dfuch) [inline src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java:275]: Are you sure about the # here? Can you double check that the generated link is working? InputStream should be {@code InputStream}. OutputStream below should be {@code OutputStream}.

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java (hunk 1 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java]: It appears as `InputStream` so I've changed `a` to `an` as suggested in new commit (3b4546e).

File: src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java (hunk 2 above)
Comment (pconcannon) [reply src/jdk.httpserver/share/classes/com/sun/net/httpserver/HttpExchange.java]: Well spotted. I have fixed the link to the `Filter` class. I also changed the first appearance of `InputStream` and `OutputSteam` to  `{@link InputStream}` and `{@link OutputStream}` and subsequent occurrences to `{@code InputStream)` and `{@code OutputStream)`. The changes can be found in commit 3b4546e

**Review by dfuch — APPROVED** [review]
Looks good to me. Alternatively you could (if you want) make sure that `InputStream` and `OutputStream` are imported and change `{@link java.io.XxxxStream}` into `{@link XxxxStream}`.

**Comment by pconcannon** [comment]
Thanks @dfuch, that's a good observation. I've incorporated it into commit 6df6ddf

**Review by dfuch — APPROVED** [review]

**Pull request closed without merging**